modules/drivers/snowflake/test/metabase/test/data/snowflake.clj

```diff
@@ -74,54 +119,91 @@
   []
   (sql-jdbc.conn/connection-details->spec :snowflake (tx/dbdef->connection-details :snowflake :server nil)))
 
-(defn- existing-dataset-names []
-  (let [db-spec (no-db-connection-spec)]
-    (jdbc/with-db-metadata [metadata db-spec]
-      ;; for whatever dumb reason the Snowflake JDBC driver always returns these as uppercase despite us making them
-      ;; all lower-case
-      (set (map u/lower-case-en (sql-jdbc.sync/get-catalogs metadata))))))
-
-(let [datasets (atom nil)]
-  (defn- existing-datasets []
-    (when-not (seq @datasets)
-      (reset! datasets (existing-dataset-names))
-      (log/infof "These Snowflake datasets have already been loaded:\n%s" (u/pprint-to-str (sort @datasets))))
-    @datasets)
-
-  (defn- add-existing-dataset! [database-name]
-    (swap! datasets conj database-name))
-
-  (defn- remove-existing-dataset! [database-name]
-    (swap! datasets disj database-name)))
+(defn- old-dataset-name?
+  "Is this dataset name prefixed by a date two days ago or older?"
+  [dataset-name]
+  (when-let [[_ year month day] (re-matches #"^(\d{4})_(\d{2})_(\d{2}).*$" dataset-name)]
+    (let [dataset-date (t/local-date (parse-long year) (parse-long month) (parse-long day))]
+      (t/before? dataset-date (u.date/add (utc-date) :day -1)))))
+
+(deftest ^:parallel old-dataset-name?-test
+  (are [s] (old-dataset-name? s)
+    "2023_02_01_82e897cb_ad31_4c82_a4b6_3e9e2e1dc1cb_test-data"
+    "2023_01_17_82e897cb_ad31_4c82_a4b6_3e9e2e1dc1cb_test-data"
+    "2022_02_17_82e897cb_ad31_4c82_a4b6_3e9e2e1dc1cb_test-data"
+    (str (unique-prefix (u.date/add (utc-date) :day -2)) "test-data"))
+  (are [s] (not (old-dataset-name? s))
+    "2050_02_17_82e897cb_ad31_4c82_a4b6_3e9e2e1dc1cb_test-data"
+    "v3_test-data"
+    (str (unique-prefix) "test-data")
+    (str (unique-prefix (u.date/add (utc-date) :day -1)) "test-data")))
+
+(defn- old-dataset-names
+  "Return a collection of all dataset names that are old -- prefixed with a date two days ago or older?"
+  []
+  (with-open [conn (jdbc/get-connection (no-db-connection-spec))]
+    (let [metadata (.getMetaData conn)]
+      (with-open [rset (.getCatalogs metadata)]
+        (loop [acc []]
```

Comment:
This is using low-level JDBC for SPEED. Not that it makes a real difference here when we're only doing this once per test run, but it was about the same amount of code to write it this way and I couldn't help myself. `clojure.java.jdbc` is pretty inefficient since it fetches every single column in the metadata result set even when we only use on.

On that note we should probably fix that stuff too 🙀

Reply:
https://github.com/metabase/metabase/pull/28422